### PR TITLE
[ci] Run only py-ci if just pydolphinscheduler change

### LINKED ISSUE #6754: [Feature][CI] Prue pydolphinscheduler change should not trigger e2e and java UT CI
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar feature requirement.

### Description

When I submit PR https://github.com/apache/dolphinscheduler/pull/6753 I found out CI `e2e` and `Java unit test` run but I only changed pydolphinscheduler code. I think it should only run `Pythons API` CI in this situation, we have `path-ignore` in `e2e` but failed and I think maybe we should use `- 'dolphinscheduler-python/pydolphinscheduler/**'` instead of `- 'dolphinscheduler-python/pydolphinscheduler'`. You could find it in https://github.com/apache/dolphinscheduler/blob/be28d2c63bb9101a3525c1102ae1a8e91f1668c9/.github/workflows/e2e.yml?_pjax=%23js-repo-pjax-container#L22

### Use case

_No response_

### Related issues

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by zhongjiajie**
closes: #6754 

This patch separate py-ci, and it only run py-ci if just python code changed

### CODE REVIEW

**Comment by zhongjiajie** [comment]
I test in my fork, it work and only `py-ci` run if only `dolphinscheduler-python/pydolphinscheduler` code change. ref https://github.com/zhongjiajie/dolphinscheduler/pull/5. PTAL @CalvinKirs @lenboo @kezhenxu94

File: .github/workflows/unit-test.yml [hunk 1]
```diff
@@ -22,12 +22,12 @@ on:
     paths-ignore:
       - '**/*.md'
       - 'dolphinscheduler-ui'
-      - 'dolphinscheduler-python/pydolphinscheduler'
+      - 'dolphinscheduler-python/pydolphinscheduler/**'
```
Comment (kezhenxu94) [inline .github/workflows/unit-test.yml:25]: if this is needed, then the line ☝️ (`'dolphinscheduler-ui'`) might also need `/**`?

Comment (zhongjiajie) [reply .github/workflows/unit-test.yml:25]: Good catch 👍 . BTW I just imitate the way `ui` but failed 🤣

**Review by kezhenxu94 — APPROVED** [review]
LGTM

**Comment by zhongjiajie** [comment]
Close this PR due to https://github.com/apache/dolphinscheduler/pull/6797, PR have to run CI named `unit test` and `build` could be merged.

**Pull request closed without merging**